src/pages/en/guides/troubleshooting.mdx

```diff
@@ -71,6 +71,31 @@ export default defineConfig({
 
 Alternatively, you can update your CSP to include `script-src: 'unsafe-inline'` to allow inline scripts to run.
 
+### exports is not defined
+
+This error can occur when you try to use a CommonJS module (CJS) in an ES module (ESM) environment.
+
+Astro uses Vite which generates [ES modules during SSR](https://vitejs.dev/guide/ssr.html#ssr-format) and [externalizes SSR dependencies](https://vitejs.dev/guide/ssr.html#ssr-externals) by default. This means that npm packages need to use the ESM format to work out-of-the-box.
+
+A common scenario is when you're trying to use an npm package that only supports the CJS format, or one of its dependencies only supports CJS.
```

Comment:
Vite does support CJS dependencies in SSR and during build, so npm packages doesn't need to use ESM. I think the common issue is that the build output could be:

```js
import { named } from 'cjs-pkg'
```

And this may sometimes not work in nodejs, which it recommends doing:

```js
import pkg from 'cjs-pkg'
const { named } = pkg
```

instead. `noExternal` does fix this case too. Perhaps we want to re-phrase the title and description about this error instead? I don't think I've seen `exports is not defined` error a lot.

Reply:
i've removed this section entirely. i'm not really sure how best to document it